### ISSUE TITLE
Segmentation Fault in Detector Simulation (Trac #2438)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix4#2430]
Running the detector simulation (`simprod-scripts/resources/scripts/detector.py`) from the latest combo release (V01-00-00 crashes with a `segmentation violation`. Most frequent, this error occurs for higher energy events (>100 TeV). This currently prevents us from running/producing simulations/MC with that combo release.

The stack trace (attachement) leads to an issue in `I3DOMLauncher`, possible connected to its recent changes for fixing the SLC timing offset of 25ns.

Chris Weaver already took a look at it and managed to track the underlaying issue down to:

```text
Assertion failed: (index>0 && index<nFADCCoarseChargeStampBins), function MakeCoarseChargeStamp, file /Users/christopher/Work/IceCube/combo/src/DOMLauncher/private/DOMLauncher/I3InIceDOM.cxx, line 166.
```

A minimum "working" example can be found in `/data/user/eganster/shared/Segfault_DOMLauncher`

<details>
 <summary><em>Migrated from <a href="https://code.icecube.wisc.edu/projects/icecube/ticket/2438">https://code.icecube.wisc.edu/projects/icecube/ticket/2438</a>, reported by egansterand owned by mjansson</em></summary>
<p>

```json
{
    "status": "closed",
    "changetime": "2020-08-13T01:48:42",
    "_ts": "1597283322257195",
    "description": "Running the detector simulation (`simprod-scripts/resources/scripts/detector.py`) from the latest combo release (V01-00-00 crashes with a `segmentation violation`. Most frequent, this error occurs for higher energy events (>100 TeV). This currently prevents us from running/producing simulations/MC with that combo release.\n\nThe stack trace (attachement) leads to an issue in `I3DOMLauncher`, possible connected to its recent changes for fixing the SLC timing offset of 25ns.\n\nChris Weaver already took a look at it and managed to track the underlaying issue down to:\n{{{\nAssertion failed: (index>0 && index<nFADCCoarseChargeStampBins), function MakeCoarseChargeStamp, file /Users/christopher/Work/IceCube/combo/src/DOMLauncher/private/DOMLauncher/I3InIceDOM.cxx, line 166.\n}}}\n\nA minimum \"working\" example can be found in `/data/user/eganster/shared/Segfault_DOMLauncher`",
    "reporter": "eganster",
    "cc": "mhuennefeld, cweaver",
    "resolution": "fixed",
    "time": "2020-08-12T12:35:02",
    "component": "combo simulation",
    "summary": "Segmentation Fault in Detector Simulation",
    "priority": "critical",
    "keywords": "",
    "milestone": "",
    "owner": "mjansson",
    "type": "defect"
}
```

</p>
</details>

**Comment by icecube-trac** [comment]
Trac update at 20200812T12:36:29: 
* eganster changed attachment from "" to "output.txt"
* eganster commented:

> DetectorSim_Output

**Comment by icecube-trac** [comment]
Trac update at 20200812T12:44:34: 
* mjansson changed status from "new" to "accepted"
* mjansson changed owner from "" to "mjansson"

**Comment by icecube-trac** [comment]
Trac update at 20200812T13:28:12: mjansson commented:

> Looking at the code there seems to be an edge case where the gaussian noise drops all the bins to below 1. The SLC code will then not find the maximum bin and return 0. 
> 
> It seems to me that the in this case the noise should have DOM not trigger a launch.

**Comment by icecube-trac** [comment]
Trac update at 20200812T16:08:53: mjansson commented:

> It seems unlikely that the noise could give this behaviour and in the particular example it crashes as the read out is all zeroes.

**Comment by icecube-trac** [comment]
Trac update at 20200813T01:45:44: cweaver commented:

> After some further investigation I have convinced myself that the all-zero waveforms triggering the crash are natural and not the result of any other issue. In both test cases which have been brought forward, they occur in the second launch on a DOM which was heavily saturated in its first launch, and thus is still severely undershot by the time it launches again. Therefore, the only fix which seems necessary is to initialize the charge stamp maximum bin index to some valid value which will either get overwritten if there is non-zero data, or which will be as good as any other index if there is not.

**Comment by icecube-trac** [comment]
Trac update at 20200813T01:48:42: 
* cweaver changed status from "accepted" to "closed"
* cweaver commented:

> In changeset 181587:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="181587"
> Initialize the charge stamp maximum bin index to a value which is valid under the new scheme. 
> This protects against bad behavior when the digitized waveform is all zero due to being undershot. 
> 
> Fixes #2430.
> ```


* cweaver changed resolution from "" to "fixed"
